app/views/Project/Tagging/Sources/SourcesFilter/EntryFilter/FrameworkFilterItem/index.tsx

```diff
@@ -0,0 +1,389 @@
+                />
+            );
+        }
+        case 'TIME_RANGE': {
+            return (
+                <TimeRangeInput
+                    className={_cs(
+                        className,
+                        styles.input,
+                        hasNoData(value?.valueGte)
+                        && hasNoData(value?.valueLte)
+                        && !allFiltersVisible
+                        && styles.hidden,
+                    )}
+                    name={name}
+                    onChange={handleTimeRangeChange}
+                    value={value?.valueGte && value?.valueLte ? {
+                        startTime: value?.valueGte,
+                        endTime: value?.valueLte,
+                    } : undefined}
+                    label={title}
+                    disabled={disabled}
+                />
+            );
+        }
+        case 'NUMBER': {
+            return (
+                <NumberButStringInput
+                    className={_cs(
+                        className,
+                        styles.input,
+                        (hasNoData(value?.value) && !allFiltersVisible)
+                        && styles.hidden,
+                    )}
+                    name="value"
+                    label={title}
+                    value={value?.value}
+                    onChange={onFieldChange}
+                    disabled={disabled}
+                />
+            );
+        }
+        case 'SCALE': {
+            return (
+                <MultiSelectInput
+                    className={_cs(
+                        className,
+                        styles.input,
+                        (hasNoData(value?.valueList) && !allFiltersVisible)
+                        && styles.hidden,
+                    )}
+                    name="valueList"
+                    value={value?.valueList}
+                    onChange={onFieldChange}
+                    label={title}
+                    options={filter.properties.options}
+                    keySelector={filterKeySelector}
+                    labelSelector={filterLabelSelector}
+                    disabled={disabled || optionsDisabled}
+                />
+            );
+        }
+        case 'GEO': {
+            return (
+                <GeoMultiSelectInput
+                    name="valueList"
+                    className={_cs(
+                        className,
+                        styles.input,
+                        (hasNoData(value?.valueList) && !allFiltersVisible)
+                        && styles.hidden,
+                    )}
+                    value={value?.valueList}
+                    onChange={onFieldChange}
+                    label={title}
+                    projectId={projectId}
+                    options={geoAreaOptions}
+                    onOptionsChange={setGeoAreaOptions}
+                    disabled={disabled}
+                    placeholder={title}
```

Comment:
We should add option to include sub-levels on geo-multiselect input

Reply:
added it but not testable right now as no GeoWidget in framework